src/main/java/org/xstefank/check/SkipChecker.java

```diff
@@ -0,0 +1,67 @@
+package org.xstefank.check;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.jboss.logging.Logger;
+import org.xstefank.model.Utils;
+import org.xstefank.model.yaml.SkipPatterns;
+import java.util.regex.Matcher;
+
+
+public class SkipChecker {
+
+    private static final Logger log = Logger.getLogger(SkipChecker.class);
+
+    private SkipPatterns skipPatterns;
+    private JsonNode payload;
+
+    public SkipChecker(SkipPatterns skipPatterns, JsonNode payload) {
+        if (payload != null ) {
+            this.payload = payload;
+        } else {
+            throw new IllegalArgumentException("Payload is null!");
+        }
+        if (skipPatterns != null) {
+            this.skipPatterns = skipPatterns;
+        } else {
+            throw new IllegalArgumentException("SkipPatterns is null!");
+        }
```

Comment:
Nothing wrong here now, but you should rather check if the params **are** null and in that case throw an exception. Probably in this case two separate if statements are redundant because you need both arguments to be present